pkg/cmd/server/api/types.go

```diff
@@ -132,15 +132,21 @@ type NodeConfig struct {
 	IPTablesSyncPeriod string
 
 	// VolumeConfig contains options for configuring volumes on the node.
-	VolumeConfig VolumeConfig
+	VolumeConfig NodeVolumeConfig
 }
 
-// VolumeConfig contains options for configuring volumes on the node.
-type VolumeConfig struct {
+// NodeVolumeConfig contains options for configuring volumes on the node.
+type NodeVolumeConfig struct {
 	// LocalQuota contains options for controlling local volume quota on the node.
 	LocalQuota LocalQuota
 }
 
+// MasterVolumeConfig contains options for configuring volumes on the node.
+type MasterVolumeConfig struct {
+	// EnableDynamicProvisioning is a boolean that toggles the dynamic provisioning feature on/off
+	EnableDynamicProvisioning bool
```

Comment:
As a general rule, controllers enablement is about sharding controllers
(admin operational performance concern).  Disabling dynamic provisioning is
an operational admin feature.

On Fri, Apr 8, 2016 at 11:20 AM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> We don't have time for the broader refactor.  It's too risky a change for
> 1.2.  DynamicProvisioning as a feature flag is ok.
>
> On Fri, Apr 8, 2016 at 11:14 AM, Jordan Liggitt <notifications@github.com>
> wrote:
>
>> In pkg/cmd/server/api/types.go
>> <https://github.com/openshift/origin/pull/8426#discussion_r59040426>:
>>
>> >  	// LocalQuota contains options for controlling local volume quota on the node.
>> >  	LocalQuota LocalQuota
>> >  }
>> >
>> > +// MasterVolumeConfig contains options for configuring volumes on the node.
>> > +type MasterVolumeConfig struct {
>> > +	// EnableDynamicProvisioning is a boolean that toggles the dynamic provisioning feature on/off
>> > +	EnableDynamicProvisioning bool
>>
>> I don't really want a one-off flag for a particular controller, I'd
>> rather work to make the controllers config param effective for disabling
>> a subset of controllers
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openshift/origin/pull/8426/files/e58a75d0de05e59a0b4fe05a06e20adc90aec313#r59040426>
>>
>
>